src/battle_script_commands.c

```diff
@@ -3503,12 +3731,20 @@ static void Cmd_getexp(void)
                     else
                         gBattleMoveDamage = 0;
 
-                    if (holdEffect == HOLD_EFFECT_EXP_SHARE)
+                    // only give exp share bonus in later gens if the mon wasn't sent out
+                    if ((holdEffect == HOLD_EFFECT_EXP_SHARE) && ((gBattleMoveDamage == 0) || (B_SPLIT_EXP < GEN_6)))
```

Comment:
Same `#ifdef` as above.

Reply:
Not used here since the if mixes defines and variables

Reply:
It could be done like this maybe:
```c
#if B_SPLIT_EXP < GEN_6
if (holdEffect == HOLD_EFFECT_EXP_SHARE && gBattleMoveDamage == 0)
#else
if (holdEffect == HOLD_EFFECT_EXP_SHARE)
#endif
```

Reply:
And which is cleaner, a five-line if or a one-line if? Ultima and I already discussed this on Discord; we both prefer the one-line form here.

Reply:
Ok